### PR TITLE
Add tests for C++ out-of-line member operators

### PR DESCRIPTION
**Pull request by MForster**
This is part of SR-12748.

### CODE REVIEW

**Comment by MForster** [comment]
@swift-ci Please test

**Comment by MForster** [comment]
@swift-ci Please test

**Comment by MForster** [comment]
@swift-ci Please test

**Comment by MForster** [comment]
@swift-ci Please test Windows platform

File: test/Interop/Cxx/operators/Inputs/member-out-of-line.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#ifndef TEST_INTEROP_CXX_OPERATORS_INPUTS_MEMBER_OUT_OF_LINE_H
+#define TEST_INTEROP_CXX_OPERATORS_INPUTS_MEMBER_OUT_OF_LINE_H
+
+struct IntBox {
+  int value;
+  IntBox operator+(IntBox rhs);
```
Comment (gribozavr) [inline test/Interop/Cxx/operators/Inputs/member-out-of-line.h:6]: Could you add a comment that this method being non-const is intentional?

File: test/Interop/Cxx/operators/member-out-of-line.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-clang -c %S/Inputs/member-out-of-line.cpp -I %S/Inputs -o %t/member-out-of-line.o -std=c++17
```
Comment (gribozavr) [inline test/Interop/Cxx/operators/member-out-of-line.swift:2]: What's C++17 specific about this test?

**Review by gribozavr — APPROVED** [review]

File: test/Interop/Cxx/operators/Inputs/member-out-of-line.h (hunk 1 above)
Comment (MForster) [reply test/Interop/Cxx/operators/Inputs/member-out-of-line.h:6]: It's not intentional. It just doesn't matter. I can make it `const` if you prefer.

File: test/Interop/Cxx/operators/member-out-of-line.swift (hunk 2 above)
Comment (MForster) [reply test/Interop/Cxx/operators/member-out-of-line.swift]: Nothing. I just copied the line. Not sure if we're targeting consistency.

**Comment by MForster** [comment]
@swift-ci Please test and merge

**Comment by MForster** [comment]
@swift-ci Please test and merge

**Comment by MForster** [comment]
@swift-ci Please test

File: test/Interop/Cxx/operators/member-out-of-line.swift (hunk 2 above)
Comment (MForster) [reply test/Interop/Cxx/operators/member-out-of-line.swift:2]: I was wrong. It's needed for the designated initializers.

**Pull request merged**